### PR TITLE
chore: add config to limit peerstore capacity

### PR DESCRIPTION
**Pull request by chaitanyaprem**
# Description

go-libp2p doesn't provide an option to limit peer-store capacity which may cause it to grow unbounded.
This PR includes a config flag which is already there in nwaku to set peerStore capacity. If not set, set it to 5*maxConnections flag.

Ref https://github.com/waku-org/nwaku/pull/1525 

# Changes

- [ ] New config flag for peer-store-capacity
- [ ] Fail adding peers in case this capacity is breached.

# Tests

Ran all unit tests.
Ran service-node with and without setting peer-store-capacity flag to confirm values are set appropriately.

Did not add any new unit tests as changes seem trivial.

### CODE REVIEW

**Review by richard-ramos — APPROVED** [review]

File: waku/v2/peermanager/peer_manager.go [hunk 1]
```diff
@@ -61,23 +63,29 @@ func inAndOutRelayPeers(relayPeers int) (int, int) {
...
 
 // NewPeerManager creates a new peerManager instance.
-func NewPeerManager(maxConnections int, logger *zap.Logger) *PeerManager {
+func NewPeerManager(maxConnections int, maxPeers int, logger *zap.Logger) *PeerManager {
 
 	maxRelayPeers, _ := relayAndServicePeers(maxConnections)
 	inRelayPeersTarget, outRelayPeersTarget := inAndOutRelayPeers(maxRelayPeers)
 
+	if maxPeers == 0 || maxConnections > maxPeers {
+		maxPeers = 5 * maxConnections
```
Comment (richard-ramos) [inline waku/v2/peermanager/peer_manager.go:72]: Maybe the `5` could be made into a constant?

**Comment by richard-ramos** [comment]
Thinking more about this PR, something that i find concerning is that peerID are ephemeral (just logout and login in desktop and you get a new peerID).  Wouldn't this be problematic? I wonder you'd end up filling slots in the peerstore with peers that might no longer exist, and reach a limit in the peerstore

**Comment by chaitanyaprem** [comment]
> Thinking more about this PR, something that i find concerning is that peerID are ephemeral (just logout and login in desktop and you get a new peerID).  Wouldn't this be problematic? I wonder you'd end up filling slots in the peerstore with peers that might no longer exist, and reach a limit in the peerstore

Good point. At some point during this I was thinking if there is a TTL for not connected peers in the peerstore. That should take care of it. If not, another approach we could take is remove peers from the store once they are disconnected, anyways now we have discv5 running constantly doing discovery which would help find peers as and when peers are less.

**Comment by chaitanyaprem** [comment]
For now, I have set address expiry of discovered peers.
When we want to connect to those peers again and addresses are not available, removing them from peerStore.
This should take care of concern you have highlighted.


Looks like we need to handle this in Waku layer in a different way. I looked at Prysm and noticed that they don't use the libp2p peerStore at all and have their own peerStore. Probably we may have to take #579 up to handle such scenarios.

**Pull request merged**